src/test/scala/parseFullTaxonomy.scala

```diff
@@ -1,67 +1,102 @@
 import ohnosequences.test.ReleaseOnlyTest
-import ohnosequences.awstools.s3.S3Object
-import java.io.File
 
-class ParseFullTaxonomy extends FunSuite {
+class ParseFullTaxonomy extends NCBITaxonomyTest("ParseFullTaxonomy") {
 
   /**
     * Auxiliary method that returns an Iterator[String] from `file`. If `file`
     * does not exist, it is downloaded from `s3Object` before parsing its lines.
     */
-  def getLines(s3Object: S3Object, file: File): Iterator[String] = {
-    if (!file.exists)
-      downloadFromS3(s3Object, file).left
-        .map { e =>
-          fail(e.msg)
-        }
-
-    retrieveLinesFrom(file) match {
-      case Right(x) => x
-      case Left(e)  => fail(e.msg)
-    }
+  def getLines(s3Object: S3Object, file: File): Lines = {
+    if (!validFile(file))
+      downloadFromS3(s3Object, file)
+
+    readLines(file)
   }
 
-  def getNamesLines(version: Version) =
+  def getNamesLines(version: Version): Lines =
     getLines(db.ncbitaxonomy.names(version), data.namesLocalFile(version))
-  def getNodesLines(version: Version) =
+
+  def getNodesLines(version: Version): Lines =
     getLines(db.ncbitaxonomy.nodes(version), data.nodesLocalFile(version))
 
   test("Parse all names and access all data", ReleaseOnlyTest) {
 
     Version.all foreach { version =>
-      dmp.names.fromLines(getNamesLines(version)) foreach { n =>
-        val id   = n.nodeID
-        val name = n.name
-
-        // We just want to check whether we can access the values but sbt
-        // complaints about the values above being unused, so trick sbt into
-        // thinkink we are using them.
-        // TODO: Code a proper test instead of this silly trick.
-        id + name
+      val seen = new HashSet[TaxID]
+
+      parse.names.fromLines(getNamesLines(version)) foreach { sciName =>
+        val id   = sciName.nodeID
+        val name = sciName.name
+
+        // Ensure only one name per id
+        assert { seen.add(id) }
+        assert { id > 0 }
+        // Ensure name is not empty
+        assert { !name.isEmpty }
       }
     }
+  }
+
+  test("All nodes can be parsed for all versions", ReleaseOnlyTest) {
 
+    Version.all foreach { version =>
+      val nodes = parse.nodes.fromLines(getNodesLines(version))
+
+      nodes foreach { maybeNode =>
+        assert { !maybeNode.isEmpty }
+      }
+    }
   }
 
-  test("Parse all nodes and access all data", ReleaseOnlyTest) {
+  test("Check that there is a name for each node", ReleaseOnlyTest) {
+
+    Version.all.foreach { version =>
+      val nodes = parse.nodes.fromLines(getNodesLines(version))
+      val names = parse.names.fromLines(getNamesLines(version))
+
+      val withName = new HashSet[TaxID]
+
+      names.foreach {
+        case ScientificName(id, name) =>
+          withName += id
+      }
+
+      nodes.foreach {
+        case Some(Node(id, _, _)) =>
+          assert { withName.contains(id) }
+        case None => // already checked that this case should not arise
```

Comment:
Mentioned test is: https://github.com/ohnosequences/db.ncbitaxonomy/blob/af40f1ea3c8d90f07521f59658cdbfc68986617d/src/test/scala/parseFullTaxonomy.scala#L45

Reply:
But if we `fail` here and we change/delete the linked test, we will still notice the error. Otherwise, we have a test that depends on the output of another test. I mean, if we wouldn't want to check that all nodes are correctly parsed, I'd agree with you, but I think it doesn't hurt to add that `fail` here to double-check.

I'd also agree with leaving the `left` out if we call the test "Check that there is a name for each correctly parsed node". Yeah, maybe this solution is better.